tests/StackTest.php

```diff
@@ -50,30 +50,32 @@ public function testServerEmptyMiddleware()
         $response = $this->getResponseMock();
 
         $stack = new Stack($response);
-        $stackResponse = $stack->process($serverRequest);
+        $stackResponse = $stack->handle($serverRequest);
 
         $this->assertInstanceOf(ResponseInterface::class, $stackResponse);
         $this->assertTrue($stackResponse === $response);
     }
 
-    public function testServerMiddlewareProcessingOrder()
+    public function testServerMiddlewareHandlingOrder()
     {
         $callCounter = 0;
 
         $middleware1 = $this->getMiddlewareMock();
         $middleware1->expects($this->once())
             ->method('process')
-            ->willReturnCallback(function (ServerRequestInterface $request, DelegateInterface $frame) use (&$callCounter
+            ->willReturnCallback(function (ServerRequestInterface $request, RequestHandlerInterface $frame) use (
```

Comment:
The $frame is repeated on multiple other places. See one here. Another one below.